code/datums/spell.dm

```diff
@@ -465,7 +470,8 @@ GLOBAL_LIST_INIT(spells, typesof(/obj/effect/proc_holder/spell))
 /obj/effect/proc_holder/spell/targeted/click/InterceptClickOn(mob/living/user, params, atom/A)
 	if(..() || !cast_check(TRUE, TRUE, user))
 		remove_ranged_ability(user)
-		revert_cast(user)
+		if (!is_casting)
```

Comment:
```suggestion
		if(!is_casting)
```
Also you never set `is_casting` to true anywhere, so it's always going to be false.